ext/opentelemetry-ext-jinja2/tests/test_jinja2.py

```diff
@@ -0,0 +1,149 @@
+        super().setUp()
+        Jinja2Instrumentor().instrument()
+        # prevent cache effects when using Template('code...')
+        # pylint: disable=protected-access
+        jinja2.environment._spontaneous_environments.clear()
+        self.tracer = get_tracer(__name__)
+
+    def tearDown(self):
+        super().tearDown()
+        Jinja2Instrumentor().uninstrument()
+
+    def test_render_inline_template(self):
+        with self.tracer.start_as_current_span("test"):
+            template = jinja2.environment.Template("Hello {{name}}!")
+            self.assertEqual(template.render(name="Jinja"), "Hello Jinja!")
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 3)
+
+        self.assertIs(spans[0].parent, spans[2].get_context())
+        self.assertIs(spans[1].parent, spans[2].get_context())
+        self.assertIsNone(spans[2].parent)
+
+        self.assertEqual(spans[0].name, "jinja2.compile")
+        self.assertIs(spans[0].kind, trace_api.SpanKind.INTERNAL)
+        self.assertEqual(
+            spans[0].attributes, {"jinja2.template_name": "<memory>"},
+        )
+
+        self.assertEqual(spans[1].name, "jinja2.render")
+        self.assertIs(spans[1].kind, trace_api.SpanKind.INTERNAL)
+        self.assertEqual(
+            spans[1].attributes, {"jinja2.template_name": "<memory>"},
+        )
+
+    def test_generate_inline_template(self):
+        with self.tracer.start_as_current_span("test"):
+            template = jinja2.environment.Template("Hello {{name}}!")
+            self.assertEqual(
+                "".join(template.generate(name="Jinja")), "Hello Jinja!"
+            )
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 3)
+
+        self.assertIs(spans[0].parent, spans[2].get_context())
+        self.assertIs(spans[1].parent, spans[2].get_context())
+        self.assertIsNone(spans[2].parent)
+
+        self.assertEqual(spans[0].name, "jinja2.compile")
+        self.assertIs(spans[0].kind, trace_api.SpanKind.INTERNAL)
+        self.assertEqual(
+            spans[0].attributes, {"jinja2.template_name": "<memory>"},
+        )
+
+        self.assertEqual(spans[1].name, "jinja2.render")
+        self.assertIs(spans[1].kind, trace_api.SpanKind.INTERNAL)
+        self.assertEqual(
+            spans[1].attributes, {"jinja2.template_name": "<memory>"},
+        )
+
+    def test_file_template(self):
+        with self.tracer.start_as_current_span("test"):
+            loader = jinja2.loaders.FileSystemLoader(TMPL_DIR)
+            env = jinja2.Environment(loader=loader)
+            template = env.get_template("template.html")
+            self.assertEqual(
+                template.render(name="Jinja"), "Message: Hello Jinja!"
+            )
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 6)
+
+        self.assertIs(spans[0].parent, spans[1].get_context())
```

Comment:
might be valuable just to unpack the spans into a semantic name.

I get the jist of what this is verifying, but it's much harder to read (especially the parent relationships) using indices.

I was thinking:

`compile_1, load_1, compile_2, load_2, render_1, test_root  = spans`

you'll get an exception anyway when you can't unpack spans into the correct count.